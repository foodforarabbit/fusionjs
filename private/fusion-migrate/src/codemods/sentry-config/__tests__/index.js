const path = require('path');
const pluginTester = require('../../../utils/codemod-test.js');

const fixtureDir = path.join(__dirname, '../__fixtures__');
const plugin = require('../plugin');

pluginTester(
  fixtureDir,
  plugin({
    common: {},
    prod: {
      clients: {
        logtron: {
          sentry: {
            id: 'http://test:test@sentry.local.uber.internal/1234',
          },
        },
      },
    },
  })
);

pluginTester(
  fixtureDir,
  plugin({
    common: {},
    prod: {
      clients: {
        logtron: {
          sentry: {
            id: 'http://test:test@sentry.uberinternal.com/1234',
          },
        },
      },
    },
  })
);

pluginTester(
  fixtureDir,
  plugin({
    common: {
      clients: {
        logtron: {
          sentry: {
            id: 'http://test:test@sentry.uberinternal.com/1234',
          },
        },
      },
    },
    prod: {},
  })
);

pluginTester(
  fixtureDir,
  plugin({
    common: {
      sentry: {
        server: {
          dsn: 'http://test:test@sentry.uberinternal.com/1234',
        },
      },
    },
    prod: {},
  })
);

pluginTester(
  fixtureDir,
  plugin({
    common: {},
    prod: {
      sentry: {
        server: {
          dsn: 'http://test:test@sentry.uberinternal.com/1234',
        },
      },
    },
  })
);
