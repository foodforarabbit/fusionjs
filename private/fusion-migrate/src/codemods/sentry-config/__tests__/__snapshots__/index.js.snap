// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sentry-config positive-match: positive-match 1`] = `
"export default {
  id: 'test',
};


      ↓ ↓ ↓ ↓ ↓ ↓

export default {id: 'http://test:test@sentry.local.uber.internal/1234'};
"
`;

exports[`sentry-config positive-match: positive-match 2`] = `
"export default {
  id: 'test',
};


      ↓ ↓ ↓ ↓ ↓ ↓

export default {id: 'http://test:test@sentry.local.uber.internal/1234'};
"
`;

exports[`sentry-config positive-match: positive-match 3`] = `
"export default {
  id: 'test',
};


      ↓ ↓ ↓ ↓ ↓ ↓

export default {id: 'http://test:test@sentry.local.uber.internal/1234'};
"
`;

exports[`sentry-config positive-match: positive-match 4`] = `
"export default {
  id: 'test',
};


      ↓ ↓ ↓ ↓ ↓ ↓

export default {id: 'http://test:test@sentry.local.uber.internal/1234'};
"
`;

exports[`sentry-config positive-match: positive-match 5`] = `
"export default {
  id: 'test',
};


      ↓ ↓ ↓ ↓ ↓ ↓

export default {id: 'http://test:test@sentry.local.uber.internal/1234'};
"
`;
